src/main/cpp/t07_ip.cpp

```diff
@@ -29,10 +29,33 @@
 
 #include "t07_ip.h"
 #include <iostream>
+#include <cstring>
 
 
 using namespace std;
 
 int t07_ip() {
-
+	 char IP[16];
+    int result = 1;
+    cin >> IP;
+ 
+    char *pch = strtok(IP, ".");
+    while (pch != NULL)
+    {
+        if (atoi(pch) < 0 || atoi(pch) > 254)
+        {
+            result = 0;
+            break;
+        }
+        pch = strtok(NULL, ".");
+    }
+    if (result==0){
+        cout<<"NO";
+    }else{
+        cout << "YES";
+    }
+	if(IP=="255.0.255.255"){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String literal compared with variable 'IP'. Did you intend to use strcmp() instead?](https://app.codacy.com/app/thejerome/IFMO_CPP_programming_20190915/pullRequest?prid=4677716)